src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +22,31 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    private static final int LETTERS_AMOUNT = 5;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        TreeSet<Character> lettersFromFile = new TreeSet<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            int input = reader.read();
+            while (input != -1) {
```

Comment:
suggestion: `while ((input = reader.read()) != -1)`

then you can remove input.read() in two places.